### PR TITLE
Refactor block module to have submodules

### PR DESCRIPTION
**Pull request by oxarbitrage**
For issue https://github.com/ZcashFoundation/zebra/issues/220

This PR deals with the break of components but do not do any of https://github.com/ZcashFoundation/zebra/issues/220#issuecomment-582644124

I was not able to understand it properly to implement neither sure if some of the requested changes are already done. Some examples of what need to be be moved to serialize will help. To keep things simple we can maybe do that in a separated PR anyway.

Please have me some patience, i am very new to rust.

### CODE REVIEW

File: zebra-chain/src/block/tests.rs [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
-use chrono::NaiveDateTime;
 use proptest::{
     arbitrary::{any, Arbitrary},
     prelude::*,
 };
 
+use crate::equihash_solution::EquihashSolution;
+use crate::merkle_tree::MerkleTreeRootHash;
+use crate::note_commitment_tree::SaplingNoteTreeRootHash;
+use crate::serialization::{ReadZcashExt, SerializationError, ZcashDeserialize, ZcashSerialize};
```
Comment (yaahc) [inline zebra-chain/src/block/tests.rs]: Once this warning is cleaned up I think this PR is good to merge

```suggestion
use crate::serialization::{SerializationError, ZcashDeserialize, ZcashSerialize};
```

**Review by yaahc — CHANGES_REQUESTED** [review]
This looks great, thank you for taking the time to clean this up for us!

**Comment by teor2345** [comment]
Using `git diff --color-moved`, I checked the code movement in this PR against the latest main branch:
* all code has been moved
* all comments have been moved
* some imports have been modified, mainly to split or merge lists (and I assume that rustc checks those)

If we make any more changes to main that break this PR, I suggest we fix them when we merge.

**Review by hdevalence — APPROVED** [review]
I committed the suggested change via the web UI to try to keep things moving just because a rearrangement change like this one where lots of things move around has a higher risk of merge conflicts the longer it stays open.

**Review by yaahc — APPROVED** [review]

**Pull request merged**